framework/mutex/OracleMutex.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\mutex;
+
+use PDO;
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * OracleMutex implements mutex "lock" mechanism via Oracle locks.
+ *
+ * Application configuration example:
+ *
+ * ```
+ * [
+ *     'components' => [
+ *         'db' => [
+ *             'class' => 'yii\db\Connection',
+ *             'dsn' => 'oci:dbname=LOCAL_XE',
+ *              ...
+ *         ]
+ *         'mutex' => [
+ *             'class' => 'yii\mutex\OracleMutex',
+ *             'lockMode' => 'NL_MODE',
+ *             'releaseOnCommit' => true,
+ *              ...
+ *         ],
+ *     ],
+ * ]
+ * ```
+ *
+ * @see Mutex
+ *
+ * @author Alexander Zlakomanov <zlakomanoff@gmail.com>
+ * @since 2.0.10
+ */
+class OracleMutex extends DbMutex
+{
+
+    /**
+     * @var array available lock modes
+     */
+    private $availableLockModes = [
+        'X_MODE',
+        'NL_MODE',
+        'S_MODE',
+        'SX_MODE',
+        'SS_MODE',
+        'SSX_MODE'
+    ];
+
+    /**
+     * @var string lock mode
+     */
+    public $lockMode = 'X_MODE';
+
+    /**
+     * @var bool release lock on commit
+     */
+    public $releaseOnCommit = false;
+
+    /**
+     * Initializes Oracle specific mutex component implementation.
+     * @throws InvalidConfigException if [[db]] is not Oracle connection.
+     */
+    public function init()
+    {
+        parent::init();
+        if (strpos($this->db->driverName, 'oci') !== 0 && strpos($this->db->driverName, 'odbc') !== 0) {
+            throw new InvalidConfigException('In order to use OracleMutex connection must be configured to use Oracle database.');
+        }
+    }
+
+    /**
+     * Acquires lock by given name.
+     * @param string $name of the lock to be acquired.
+     * @param integer $timeout to wait for lock to become released.
+     * @return boolean acquiring result.
+     * @see http://docs.oracle.com/cd/B19306_01/appdev.102/b14258/d_lock.htm
+     */
+    protected function acquireLock($name, $timeout = 0)
+    {
+        $lockStatus = null;
+
+        /** clean vars before using */
+        $releaseOnCommit = ($this->releaseOnCommit === true) ? 'TRUE' : 'FALSE';
+        $timeout = abs((int)$timeout);
+
+        if(!in_array($this->lockMode, $this->availableLockModes)){
```

Comment:
If you about releaseOnCommit parameter - this parameter false by default
http://docs.oracle.com/cd/B19306_01/appdev.102/b14258/d_lock.htm#i1002556